app/controllers/responses_controller.rb

```diff
@@ -168,16 +170,19 @@ def web_create_or_update
   end
 
   def handle_odk_submission
-    return render_xml_submission_failure("No XML file attached.", 422) unless params[:xml_submission_file]
+    submission_file = params[:xml_submission_file]
+    return render_xml_submission_failure("No XML file attached.", 422) unless submission_file
 
     # See config/initializers/http_status_code.rb for custom status definitions
     begin
+      tmp_path = copy_to_tmp_path(submission_file)
+
       @response.user_id = current_user.id
       @response.device_id = params[:deviceID]
-      @response.odk_xml = params[:xml_submission_file]
+      @response.odk_xml = submission_file
       @response = odk_response_parser.populate_response
       authorize!(:submit_to, @response.form)
-      @response.save(validate: false)
+      FileUtils.rm(tmp_path) if @response.save(validate: false)
```

Comment:
I think you can just do 

```
@response.save(validate: false)
FileUtils.rm(tmp_path)
```

`save` doesn't return anything meaningful if called with `validate: false`